dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java

```diff
@@ -0,0 +1,315 @@
+	
+	private static final Logger log = Logger.getLogger(WorkspaceItemRestRepository.class);
+	
+	@Autowired
+	WorkspaceItemService wis;
+	@Autowired
+	ItemService itemService;
+	@Autowired
+	BitstreamService bitstreamService;
+	@Autowired
+	BitstreamFormatService bitstreamFormatService;
+	@Autowired
+	ConfigurationService configurationService;
+	
+	@Autowired
+	WorkspaceItemConverter converter;
+	
+	@Autowired
+	SubmissionService submissionService;
+	@Autowired
+	EPersonServiceImpl epersonService;
+	
+	private SubmissionConfigReader submissionConfigReader;
+	
+	public WorkspaceItemRestRepository() throws SubmissionConfigReaderException {
+		submissionConfigReader = new SubmissionConfigReader();
+	}
+
+	@Override
+	public WorkspaceItemRest findOne(Context context, Integer id) {
+		WorkspaceItem witem = null;
+		try {
+			witem = wis.find(context, id);
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		if (witem == null) {
+			return null;
+		}
+		return converter.fromModel(witem);
+	}
+
+	@Override
+	public Page<WorkspaceItemRest> findAll(Context context, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {
+			total = wis.countTotal(context);
+			witems = wis.findAll(context, pageable.getPageSize(), pageable.getOffset());
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<WorkspaceItemRest> page = new PageImpl<WorkspaceItem>(witems, pageable, total).map(converter);
+		return page;
+	}
+	
+	@SearchRestMethod(name = "findBySubmitter")
+	public Page<WorkspaceItemRest> findBySubmitter(@Param(value="uuid") UUID submitterID, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {		
+			Context context = obtainContext();
+			EPerson ep = epersonService.find(context, submitterID); 
+			witems = wis.findByEPerson(context, ep);
+			total = witems.size();
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<WorkspaceItemRest> page = new PageImpl<WorkspaceItem>(witems, pageable, total).map(converter);
+		return page;
+	}
+	
+	@Override
+	protected WorkspaceItemRest createAndReturn(Context context) {
+		WorkspaceItem source = submissionService.createWorkspaceItem(context, getRequestService().getCurrentRequest());		
+		return converter.convert(source);		
+	}
+
+	@Override
+	protected WorkspaceItemRest save(Context context, WorkspaceItemRest wsi) {
+		SubmissionConfig submissionConfig = submissionConfigReader.getSubmissionConfigByName(submissionConfigReader.getDefaultSubmissionConfigName());
+		WorkspaceItem source = converter.toModel(wsi);
+		for(int stepNum = 0; stepNum<submissionConfig.getNumberOfSteps(); stepNum++) {
+			
+			SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+			/*
+			 * First, load the step processing class (using the current
+			 * class loader)
+			 */
+			ClassLoader loader = this.getClass().getClassLoader();
```

Comment:
This code is a duplicated from the code in `WorkspaceItemConverter`. Is there a way to not duplicate that code? (Either move it to the converter class or to the rest model class)

Reply:
As above. Note that we are thinking to migrate the submission configuration as Spring bean. All pieces of code like this will be deleted and rewrited.